tensorboard/plugins/pr_curve/__init__.py

```diff
@@ -0,0 +1,195 @@
+
+    # First compute the bucket indices for each prediction value.
+    bucket_indices = (
+        tf.cast(tf.floor(predictions * (num_thresholds-1)), tf.int32))
+    # Adjust indicies by classes. For performance and simplicity, we keep
+    # the buckets (see below) as 1D array representing the tp/fp buckets for
+    # a (num_classes, num_thresholds) tensors.
+    # So, for each index in bucket_indices, its real index into this 1D array is
+    #   index + (num_thresholds * class_index)
+    class_indices = tf.constant(range(num_classes), shape=(1, num_classes))
+    bucket_indices += num_thresholds * class_indices
+
+    with tf.name_scope('variables'):
+      # Now create the variables which correspond to the bucket values.
+      # These are flat arrays with num_thresholds values per class.
+      tp_buckets_v = tf.contrib.framework.local_variable(
+          tf.zeros(
+              [num_classes * num_thresholds],
+              dtype=dtype),
+          name='tp_buckets')
+      fp_buckets_v = tf.contrib.framework.local_variable(
+          tf.zeros(
+              [num_classes * num_thresholds],
+              dtype=dtype),
+          name='fp_buckets')
+
+    initialize_bucket_counts = tf.variables_initializer(
+        [tp_buckets_v, fp_buckets_v])
+    with tf.control_dependencies([initialize_bucket_counts]):
+      # Create the non-flat views with class_index as first dimension.
+      tp_buckets = tf.reshape(tp_buckets_v, (num_classes, -1))
+      fp_buckets = tf.reshape(fp_buckets_v, (num_classes, -1))
+
+      with tf.name_scope('update_op'):
+        # Use scatter_add to update the buckets.
+        update_tp = tf.scatter_add(
+            tp_buckets_v, bucket_indices, true_labels, use_locking=True)
+        update_fp = tf.scatter_add(
+            fp_buckets_v, bucket_indices, false_labels, use_locking=True)
+
+    with tf.control_dependencies([update_tp, update_fp]):
+      with tf.name_scope('metrics'):
+        thresholds = tf.constant(
+            np.linspace(0.0, 1.0, num_thresholds), dtype=dtype)
+        # Set up the cumulative sums to compute the actual metrics.
+        tp = tf.cumsum(tp_buckets, reverse=True, axis=1, name='tp')
+        fp = tf.cumsum(fp_buckets, reverse=True, axis=1, name='fp')
+        # fn = sum(true_labels) - tp
+        #    = sum(tp_buckets) - tp
+        #    = tp[:, 0] - tp
+        # Similarly,
+        # tn = fp[:, 0] - fp
+        tn = tf.subtract(fp[:, 0:1], fp, name='tn')
+        fn = tf.subtract(tp[:, 0:1], tp, name='tp')
+
+        # Store the number of thresholds within the summary metadata because
+        # that value is constant for all pr curve summaries with the same tag.
+        summary_metadata = tf.SummaryMetadata()
+        pr_curve_plugin_data = _PrCurvePluginData(num_thresholds=num_thresholds)
+        summary_metadata.plugin_data.add(
+            plugin_name=constants.PLUGIN_NAME,
+            content=json.dumps(pr_curve_plugin_data._asdict()))
```

Comment:
Consider `content=json.dumps({'num_thresholds': num_thresholds})`. I'm assuming you added the namedtuple to have structure, but I think namedtuple isn't really good at adding structure to JSON.

In fact, what I would recommend is defining a protobuf. Then put the binary serialized content in the summary_metadata field. That would:

1. Give you an object with a schema
2. Use less space on disk.

The event logs are already stored as a binary protobuf, so why not put more binary protobuf in there?